test/cli.js

```diff
@@ -680,6 +680,49 @@ test('legacy snapshot files are reported to the console', t => {
 	});
 });
 
+test('snapshots infer their location from sourcemaps', t => {
+	t.plan(8);
+	const dirname = path.join('fixture/snapshots/test-sourcemaps');
+	const testLocationTypes = [
+		'src',
+		'src/test/snapshots',
+		'src/feature/__tests__/__snapshots__'
+	];
+	const removeExists = relFilePath => {
```

Comment:
Yes this is wrong, I have updated this be a little more dry https://github.com/avajs/ava/pull/1489/commits/12f1ff8d73e0ae47c109eed9737470c9d421fa8e#diff-583d616f4b01bf1fa153e71c0cef7de7R704. If there are more integration tests on projects like this I can think of some better ways to do it, like maybe the project should be first duplicated to a /tmp location and so on.